internal/web/common/endpoint.go

```diff
@@ -0,0 +1,95 @@
+package common
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/eurofurence/reg-room-service/internal/apierrors"
+	"github.com/eurofurence/reg-room-service/internal/logging"
+)
+
+type (
+	RequestHandler[Req any]  func(r *http.Request) (*Req, error)
```

Comment:
I don't think this is a good practice. Error handling on Go should be abstracted with the error interface. Making use of error functions like `Errorf` or `errors.Wrap(f)` wouldn't be possible anymore.

I recommend letting functions return a generic error interface and let the consumer decide how to react to it. Feels more like the "Go way of doing it" to me.

Reply:
I disagree in this specific instance, because it makes the compiler help me ensure I always send an ErrorDto and think about what http status and message to send, as demanded by the service's OpenAPI spec. Anything where the compiler helps me avoid mistakes, I want to utilize. That way I can't forget to wrap errors and assign a http status to them, and every 500 I send is a conscious decision (or the result of a panic).

Of course it shouldn't be *StatusError, but *APIError, I confused those two.